pkg/internal/session/session.go

```diff
@@ -29,6 +29,8 @@ type Options struct {
 	RouteExp       string            // expression of how to route the traffic to the target resource
 	Strategy       string            // name of the strategy to use for the target resource
 	StrategyArgs   map[string]string // additional arguments for the strategy
+	Revert         bool              // Revert back to previous known value if join/leave a existing session with a known ref
+	Client         *Client           // Alternative client to be used. If nil a DefaultClient is created
```

Comment:
I am not sure `Client` should belong to this data structure, to be honest. Can we simply pass it to `Handler` when creating it?

Reply:
both client and handler were private structs and are not normally known outside. I found it more sensible to be able to swap the Client impl then the Handler impl.

Reply:
My point was more about where the `Client` instance should belong. `Options` feels to me like plain data structure and holding an instance of the `Client` makes a bit more than it should be in my eyes. Thus my thinking was that `Client` could be passed when we call `CreateOrJoin` or any other func which creates instance of the `Handler`.

Reply:
Ah, sorry, I mixed Handler and handler :)

We can make it `type Handler func(opts Options, client Client) (State, func(), error)` for sure.

Reply:
https://github.com/maistra/istio-workspace/pull/492/commits/9b531bd991705e7e12689f5fd4f45c966730c78a